core/app/models/spree/tax/tax_location.rb

```diff
@@ -25,6 +25,10 @@ def ==(other)
         state_id == other.state_id && country_id == other.country_id
       end
 
+      def country
+        Spree::Country.find_by(id: country_id)
```

Comment:
I wouldn't do it. The `Spree::Tax::TaxLocation` is usually a pretty throwaway object. The big reason why we need this method is that countries are referenced by ID rather that ISO code (in which case I could just use that and not use a DB lookup). 

Also: This query can return `nil`, in which case memoizing won't help.